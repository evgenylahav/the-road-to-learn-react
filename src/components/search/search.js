import * as React from 'react';

export class Search extends React.Component {
  componentDidMount(){
    if (this.input) {
      this.input.focus();
    }
  }

  render() {
    const { value, onChange, onSubmit, children } = this.props;
    
    return(
        <div data-test="search-form">
          <form onSubmit={onSubmit}>
            {children}<input
              type="text"
              value={value}
              onChange={onChange}
              ref={el => this.input = el}
            />
            <button type="submit">
              {children}
            </button>
          </form>
        </div>
    );
  }
}
