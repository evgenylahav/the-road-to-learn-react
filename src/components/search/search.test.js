import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Search } from './search';
import { shallow } from 'enzyme';
import { testNameToKey } from 'jest-snapshot/build/utils';

describe('Search', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('creates a search form', () => {
    const element = shallow(
      <Search>Search</Search>
    );
    expect(element.find(`[data-test='search-form']`).length).toBe(1);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Search>Search</Search>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
