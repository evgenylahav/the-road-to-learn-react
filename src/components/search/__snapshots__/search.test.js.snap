// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<div
  data-test="search-form"
>
  <form>
    Search
    <input
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
